### PR TITLE
Fix for iOS project generation from root directory

### PR DESCRIPTION
**Pull request by lucidsheep**
Resolves https://github.com/facebook/buck/issues/468

### CODE REVIEW

**Comment by bhamiltoncx** [comment]
Very nice! Could you possibly add an integration test that covers this case?

**Comment by lucidsheep** [comment]
Added an integration test - fails on master and passes on this branch. Let me know if it looks good!

**Comment by Coneko** [comment]
cc @jspahrsummers, @ndfred

File: src/com/facebook/buck/apple/SchemeGenerator.java [hunk 1]
```diff
@@ -130,10 +130,20 @@ public Path writeScheme() throws IOException {
       if (blueprintName == null) {
         blueprintName = target.getName();
       }
+      Path outputPath = outputDirectory.getParent();
+      String buildableReferencePath;
+      if(outputPath == null) {
+          //Root directory project
+          outputPath = outputDirectory;
```
Comment (Coneko) [inline src/com/facebook/buck/apple/SchemeGenerator.java]: This doesn't seem necessary.

File: test/com/facebook/buck/apple/xcode/ProjectIntegrationTest.java [hunk 2]
```diff
@@ -182,6 +182,23 @@ public void generatingCombinedProject() throws IOException {
...
+        "generating_root_directory_project",
+        temporaryFolder);
+    workspace.setUp();
+
+    ProjectWorkspace.ProcessResult result = workspace.runBuckCommand(
+        "project",
+        "//:bundle");
+    result.assertSuccess();
+
+    workspace.verify();
```
Comment (Coneko) [inline test/com/facebook/buck/apple/xcode/ProjectIntegrationTest.java]: This isn't doing anything: you don't have any `.expected` files in your test data.

File: test/com/facebook/buck/apple/xcode/ProjectIntegrationTest.java [hunk 3]
```diff
@@ -182,6 +182,23 @@ public void generatingCombinedProject() throws IOException {
...
+        this,
+        "generating_root_directory_project",
+        temporaryFolder);
+    workspace.setUp();
+
+    ProjectWorkspace.ProcessResult result = workspace.runBuckCommand(
+        "project",
+        "//:bundle");
+    result.assertSuccess();
+
```
Comment (Coneko) [inline test/com/facebook/buck/apple/xcode/ProjectIntegrationTest.java]: Add some asserts on where the generated project, workspace and schemes should be.

**Comment by Coneko** [comment]
Looks good, just a couple of notes.

**Comment by lucidsheep** [comment]
Thanks for the review @Coneko . I added some .expected files for the generated project.

**Comment by Coneko** [comment]
@facebook-github-bot shipit

**Comment by bhamiltoncx** [comment]
Well done, by the way! This is one of those really nasty long-standing bugs that affects people just getting started with Buck. I'm really glad you fixed it.

**Pull request closed without merging**